comp/dogstatsd/server/parse.go

```diff
@@ -238,8 +246,25 @@ func (p *parser) parseFloat64List(rawFloats []byte) ([]float64, error) {
 }
 
 // extractContainerID parses the value of the container ID field.
+// If the field is prefixed by `in-`, it corresponds to the cgroup controller's inode of the source
+// and is used for ContainerID resolution.
 func (p *parser) extractContainerID(rawContainerIDField []byte) []byte {
-	return rawContainerIDField[len(containerIDFieldPrefix):]
+	containerIDField := rawContainerIDField[len(containerIDFieldPrefix):]
+
+	if bytes.HasPrefix(containerIDField[:len(containerIDFieldInodePrefix)], containerIDFieldInodePrefix) {
+		inodeField, err := strconv.ParseUint(string(containerIDField[len(containerIDFieldPrefix)+1:]), 10, 64)
+		if err != nil {
+			return nil
+		}
+
+		containerID, err := provider.GetProvider().GetMetaCollector().GetContainerIDForInode(inodeField, readerCacheExpiration)
+		if err != nil {
+			return nil
+		}
+		return []byte(containerID)
```

Comment:
I might be wrong, but don't you want to test for `if containerID == ""` in the case the metacollector was not able to return a container ID with the given inode id? I think this method should return `nil` instead of an empty `[]byte`, WDYT?

Reply:
The current implementation actually check if the tag is an empty string and will drop it if that's the case.